src/bert_model.rs

```diff
@@ -192,6 +193,54 @@ impl LoadFromHDF5 for BertEmbeddings {
     }
 }
 
+#[derive(Debug)]
+pub struct BertEncoder {
+    layers: Vec<BertLayer>,
+}
+
+impl BertEncoder {
+    /// Apply the encoder.
+    ///
+    /// Returns the hidden states and attention per layer.
+    pub fn forward_t(&self, hidden_states: &Tensor, train: bool) -> Vec<(Tensor, Tensor)> {
+        let mut all_hidden_states = Vec::with_capacity(self.layers.len());
+
+        let mut hidden_states = CowTensor::Borrowed(hidden_states);
+        for layer in &self.layers {
+            let states_attention = layer.forward_t(&hidden_states, train);
+
+            hidden_states = CowTensor::Owned(states_attention.0.shallow_clone());
```

Comment:
Not this PR, should have been in the previous review - maybe the return type of BertLayer should have named fields?